java/dataset/pom.xml

```diff
@@ -25,6 +25,8 @@
     <arrow.cpp.build.dir>../../../cpp/release-build/</arrow.cpp.build.dir>
     <parquet.version>1.13.1</parquet.version>
     <avro.version>1.11.3</avro.version>
+    <substrait.version>0.31.0</substrait.version>
+    <protobuf.version>3.25.3</protobuf.version>
```

Comment:
No, this is only defined in gandiva like this format, not exists in  dependencyManagement

Reply:
Hmm, I won't force it here. @vibhatha that would be an idea for future cleanup though - try to make sure we use dependencyManagement and BOM (where available) for all dependencies instead of declaring dependencies ad-hoc inside each individual module.

Reply:
Issue created https://github.com/apache/arrow/issues/42205
I will work on this as soon as this PR is merged.